server/public/model/config.go

```diff
@@ -4519,6 +4519,63 @@ func (o *Config) Sanitize(pluginManifests []*Manifest) {
 	o.PluginSettings.Sanitize(pluginManifests)
 }
 
+type FilterTag struct {
+	TypeOfTag string
+	Tag       string
+}
+
+type ConfigFilterOptions struct {
+	TagFilters     []FilterTag
+	RemoveMasked   bool
+	RemoveDefaults bool
+}
+
+type GetConfigOptions struct {
+	RemoveMasked   bool
+	RemoveDefaults bool
+}
+
+func FilterConfig(cfg *Config, opts ConfigFilterOptions) (map[string]any, error) {
```

Comment:
I am inclined towards having this function inside app/config.go. This feels like app specific logic.

Reply:
It's actually a handy thing as a utility function, since all filtering was defined here I thought keeping here would be more cohesive. WDYT?

Reply:
In that case, could we add a comment to the function clearly explaining its function? Thanks.